sklearn/datasets/_base.py

```diff
@@ -1487,14 +1490,33 @@ def _fetch_remote(remote, dirname=None):
     dirname : str
         Directory to save the file to.
 
+    n_retries : int
+        Number of retries when HTTP errors are encountered.
+
+    delay : int
+        Number of seconds between retries.
+
     Returns
     -------
     file_path: str
         Full path of the created file.
     """
 
     file_path = remote.filename if dirname is None else join(dirname, remote.filename)
-    urlretrieve(remote.url, file_path)
+
+    retry_cnt = n_retries
+    while 1:
+        try:
+            urlretrieve(remote.url, file_path)
+            break
+        except URLError:
```

Comment:
Extending from https://github.com/scikit-learn/scikit-learn/pull/22529 , lets also retry for `TimeoutError`:

```suggestion
        except (URLError, TimeoutError):
```